src/databricks/sql/ae.py

```diff
@@ -111,12 +115,25 @@ def _thrift_cancel_operation(self) -> None:
         _output = self._thrift_backend.async_cancel_command(self.t_operation_handle)
         self.status = AsyncExecutionStatus.CANCELED
 
+    def serialize(self) -> str:
+        """Return a string representing the query_id and secret of this AsyncExecution.
+
+        Use this to preserve a reference to the query_id and query_secret."""
+        return f"{self.query_id}:{self.query_secret}"
+
     def poll_for_status(self) -> None:
         """Check the thrift server for the status of this operation and set self.status
 
         This will result in an error if the operation has been canceled or aborted at the server"""
 
-        _output = self._thrift_backend._poll_for_status(self.t_operation_handle)
+        try:
```

Comment:
Perfect example of why I don't love complexity in my constructors.  An ideal constructor says, if you give me all the required parameters as input, you will get an instance of this object as object; here, however, our constructor could fail and throw an exception to the user.  If you instead use a factory pattern, you have the choice of propagating the error, or just giving the user None, as you could name the method something like 'get_if_exists(...)'.

Reply:
That makes sense to me.

I _do_ have a factory function. It's the `Connection.get_async_execution()` method. I can push this "does it exist" checking into that function like you describe.